src/nvim/screen.c

```diff
@@ -3300,6 +3050,10 @@ win_line (
         p_extra++;
       }
       n_extra--;
+    } else if (foldinfo.fi_lines > 0) {
```

Comment:
this part and display of buffer is a bit hackish/ugly. Seems to work and @bfredl as you mentioned we may use virttext later. As perfect is the worst enemy of good, I vote for merging any way: I've tested locally and it works ok.

Reply:
btw thanks for completing this, super cool